artiq/frontend/artiq_sinara_tester.py

```diff
@@ -244,6 +253,63 @@ def test_ttl_ins(self):
             else:
                 print("FAILED")
 
+    def test_lvds_group(self, ttl_out_chunk, ttl_in_chunk):
+        def devices_to_string(chunk):
+            return ', '.join(map(lambda x: x[0], chunk))
+
+        input(f"\nConnect ({devices_to_string(ttl_out_chunk)}) group to ({devices_to_string(ttl_in_chunk)}) group . Press ENTER when done.")
+        for x in range(4):
+            ttl_in, ttl_in_dev = ttl_in_chunk[x]
+            ttl_out, ttl_out_dev = ttl_out_chunk[x]
+            print(f"Testing {ttl_in} with {ttl_out}... ", end="")
+
+            if self.test_ttl_in(ttl_out_dev, ttl_in_dev):
+                print("PASSED")
+            else:
+                print("FAILED")
+
+    def test_ttl_lvds(self):
+        def lower_bound_of_device_chunk(dev_list, dev_name, N):
+            dev_index = next(i for i, v in enumerate(dev_list) if v[0] == dev_name)
+            return dev_index - (dev_index % N)
+
+        print(f"*** Testing LVDS TTL")
+
+        if len(self.ttl_lvds_ins) == 0:
+            print(f"No available input channels for testing. Skipping...")
+            return
+
+        ttl_in_index, ttl_out_index = 0, -1
+        default_ttl_in_name, _ = self.ttl_lvds_ins[ttl_in_index]
+        default_ttl_out_name, _ = self.ttl_lvds[ttl_out_index]
+
+        print("LVDS TTL channels are tested in groups of 4.")
+        print("You can choose any channel within the group.")
```

Comment:
Would it be better to specify the group (channel % 4), or have it "align" to a group?

Reply:
It already does it (?) - if you specify any channel in group `ttl0...ttl3`, it will select that group.

Reply:
As in, instead of asking for a channel, it would ask for a group (instead of ttl3, group 0). Especially here, where a unit of test is essentially a group anyway - you connect the RJ45 cable to one. Then you wouldn't have to calculate the lower boundary either.

It is a bit of a nitpick and functionally wouldn't change anything, but I would consider it being a bit confusing at a first glance.

Reply:
I thought about it, but decided that it would confuse the tester more.